platforms/jvm/toolchains-jvm/src/main/java/org/gradle/jvm/internal/services/ToolchainsJvmServices.java

```diff
@@ -41,15 +40,17 @@
 import org.gradle.jvm.toolchain.internal.install.DefaultJavaToolchainProvisioningService;
 import org.gradle.jvm.toolchain.internal.install.JdkCacheDirectory;
 import org.gradle.jvm.toolchain.internal.install.SecureFileDownloader;
+import org.gradle.platform.Architecture;
 import org.gradle.platform.internal.DefaultBuildPlatform;
 
 import java.util.List;
 
 public class ToolchainsJvmServices extends AbstractPluginServiceRegistry {
     protected static class BuildServices {
 
-        protected DefaultBuildPlatform createBuildPlatform(ObjectFactory objectFactory, SystemInfo systemInfo, OperatingSystem operatingSystem) {
-            return objectFactory.newInstance(DefaultBuildPlatform.class, systemInfo, operatingSystem);
+        protected DefaultBuildPlatform createBuildPlatform(ObjectFactory objectFactory, OperatingSystem operatingSystem) {
+            Architecture architecture = Architecture.current();
```

Comment:
We should add a provider for `Architecture` in `NativeServices` so we can inject the current arch like we do with `OperatingSystem`:

https://github.com/gradle/gradle/blob/b466244a613f6649df17e32a323f1fe43dad4d2c/platforms/core-runtime/native/src/main/java/org/gradle/internal/nativeintegration/services/NativeServices.java#L358-L361